src/main/java/it/pagopa/transactions/services/TransactionsService.java

```diff
@@ -336,164 +350,195 @@ public Mono<TransactionInfoDto> updateTransactionAuthorization(
-                                                                            )
-                                                                    )
-                                                                    .flatMap(
-                                                                            authorizationStatusUpdatedEvent -> authorizationUpdateProjectionHandler
-                                                                                    .handle(
-                                                                                            authorizationStatusUpdatedEvent
-                                                                                    )
-                                                                    );
-                                                        }
-                                                )
-                                                .cast(TransactionActivated.class)
-                                                .flatMap(
-                                                        transaction -> {
-                                                            ClosureSendData closureSendData = new ClosureSendData(
-                                                                    transaction,
-                                                                    updateAuthorizationRequestDto
-                                                            );
-
-                                                            TransactionClosureSendCommand transactionClosureSendCommand = new TransactionClosureSendCommand(
-                                                                    transaction.getPaymentNotices().get(0).rptId(),
-                                                                    closureSendData
-                                                            );
-
-                                                            return transactionSendClosureHandler
-                                                                    .handle(transactionClosureSendCommand)
-                                                                    .doOnNext(
-                                                                            closureSentEvent ->
-                                                            // FIXME Handle multiple rtpId
-                                                            log.info(
-                                                                    "Requested transaction closure for rptId: {}",
-                                                                    transaction.getPaymentNotices().get(0).rptId()
-                                                                            .value()
+                                                        );
+                                                    }
+                                            );
+
+                                        }
+                                    })
```

Comment:
Alternative version, more of a matter of style (replaced direct `Either` constructor call with call through `map`):

```suggestion
                                    .<Either<TransactionInfoDto, Mono<BaseTransaction>>>flatMap(alreadyAuthorized -> {
                                        if (!alreadyAuthorized) {
                                            return Mono.just(baseTransaction).map(Either::right);
                                        } else {
                                            return baseTransaction.map(
                                                    trx -> {
                                                        log.info(
                                                                "Transaction authorization outcome already received. Transaction status: {}",
                                                                trx.getStatus()
                                                        );
                                                        return buildTransactionInfoDto(trx);
                                                    }
                                            ).map(Either::left);
```

Reply:
Fixed with https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/168/commits/abfdb3480ce8cb9b2d53ad44070cbd6535e71c4c